### ISSUE TITLE
Maven metadata sometime incorrect between deployment and test

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#10534]
The build sometimes fails with errors like this:

```
[INFO] Skipping artifact installation
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building Spring Boot Data Redis Sample 2.0.0.BUILD-SNAPSHOT
[INFO] ------------------------------------------------------------------------
Downloading: http://repo.spring.io/snapshot/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/maven-metadata.xml
Downloading: file:///tmp/build/fd4135ff/distribution-repository/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/maven-metadata.xml
Downloaded: file:///tmp/build/fd4135ff/distribution-repository/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/maven-metadata.xml (1.0 kB at 116 kB/s)
Downloaded: http://repo.spring.io/snapshot/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/maven-metadata.xml (1.0 kB at 4.2 kB/s)
Downloading: file:///tmp/build/fd4135ff/distribution-repository/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023014-1.pom
Downloading: http://repo.spring.io/snapshot/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023014-1.pom
[WARNING] The POM for org.springframework.boot:spring-boot-starter-data-redis:jar:2.0.0.BUILD-20171006.023014-1 is missing, no dependency information available
Downloading: file:///tmp/build/fd4135ff/distribution-repository/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023014-1.jar
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Spring Boot Samples ................................ SUCCESS [  3.094 s]
[INFO] Spring Boot Ant Sample ............................. SUCCESS [ 32.001 s]
<snip>
[INFO] Spring Boot XML Sample ............................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:38 min
[INFO] Finished at: 2017-10-06T02:42:54Z
[INFO] Final Memory: 459M/849M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project spring-boot-sample-data-redis: Could not resolve dependencies for project org.springframework.boot:spring-boot-sample-data-redis:jar:2.0.0.BUILD-SNAPSHOT: Could not find artifact org.springframework.boot:spring-boot-starter-data-redis:jar:2.0.0.BUILD-20171006.023014-1 in repository (file:///tmp/build/fd4135ff/distribution-repository) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :spring-boot-sample-data-redis
```

The root cause seems to be a mix up in the meta-data file:

```
$ cat /tmp/build/fd4135ff/distribution-repository/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT/maven-metadata.xml
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<metadata modelVersion="1.1.0">
  <groupId>org.springframework.boot</groupId>
  <artifactId>spring-boot-starter-data-redis</artifactId>
  <version>2.0.0.BUILD-SNAPSHOT</version>
  <versioning>
    <snapshot>
      <timestamp>20171006.023014</timestamp>
      <buildNumber>1</buildNumber>
    </snapshot>
    <lastUpdated>20171006023014</lastUpdated>
    <snapshotVersions>
      <snapshotVersion>
        <extension>jar</extension>
        <value>2.0.0.BUILD-20171006.023014-1</value>
        <updated>20171006023014</updated>
      </snapshotVersion>
      <snapshotVersion>
        <extension>pom</extension>
        <value>2.0.0.BUILD-20171006.023014-1</value>
        <updated>20171006023014</updated>
      </snapshotVersion>
      <snapshotVersion>
        <classifier>sources</classifier>
        <extension>jar</extension>
        <value>2.0.0.BUILD-20171006.023013-1</value>
        <updated>20171006023014</updated>
      </snapshotVersion>
    </snapshotVersions>
  </versioning>
</metadata>
```

```
$ ls -l /tmp/build/fd4135ff/distribution-repository/org/springframework/boot/spring-boot-starter-data-redis/2.0.0.BUILD-SNAPSHOT
total 48
-rw-r--r-- 1 root root 1042 Oct  6 02:30 maven-metadata.xml
-rw-r--r-- 1 root root   32 Oct  6 02:30 maven-metadata.xml.md5
-rw-r--r-- 1 root root   40 Oct  6 02:30 maven-metadata.xml.sha1
-rw-r--r-- 1 root root 2390 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1-sources.jar
-rw-r--r-- 1 root root   32 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1-sources.jar.md5
-rw-r--r-- 1 root root   40 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1-sources.jar.sha1
-rw-r--r-- 1 root root 2576 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1.jar
-rw-r--r-- 1 root root   32 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1.jar.md5
-rw-r--r-- 1 root root   40 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1.jar.sha1
-rw-r--r-- 1 root root 2714 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1.pom
-rw-r--r-- 1 root root   32 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1.pom.md5
-rw-r--r-- 1 root root   40 Oct  6 02:30 spring-boot-starter-data-redis-2.0.0.BUILD-20171006.023013-1.pom.sha1
root@c3d185f5-cbac-45ba-5d5a-f2874258f9c3:/tmp/build/fd4135ff# 
```

**Comment by philwebb** [comment]
Quite possibly https://issues.apache.org/jira/browse/MDEPLOY-221?focusedCommentId=16049493&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16049493

**Comment by philwebb** [comment]
Hopefully fixed by the Maven upgrade in e79db1d9f57b57f884ee216aee159c8d84e4144a
